eZ/Publish/Core/Repository/SectionService.php

```diff
@@ -53,6 +58,7 @@ class SectionService implements SectionServiceInterface
     public function __construct(RepositoryInterface $repository, Handler $sectionHandler, array $settings = array())
     {
         $this->repository = $repository;
+        $this->permissionResolver = $repository->getPermissionResolver();
```

Comment:
Rather just a question regarding these changes. In all other services `$this->repository->canUser()` is used, is `SectionService` different somehow and needs to has an instance of `PermissionResolver`?

Reply:
@kmadejski https://github.com/ezsystems/ezpublish-kernel/blob/4e0ac666a8bf2d3988019e203831b059e6548db6/eZ/Publish/API/Repository/Repository.php#L61

Reply:
@wizhippo thanks! I know about that, this is exactly why I've asked this question. As you can see, even after changes from this PR all other services still use deprecated `$this->repository->canUser()`. Therefore, I'm just curious why this has been changed only in `SectionService`.

Reply:
Got it.  Good question.

Reply:
@kmadejski because it feels a bit out of the scope of this PR. Too many changes slow down review and approval. 
But you are more than welcomed to improve our code which still uses `Repository::canUser` directly :grinning:

Reply:
So yeah, I did it on this one only as here it was not to invasive, but in general it's a small step we should do all over to help us be able to de-couple this at later point. As it makes it more clear what is using repo as this is the number one thing we want to get rid of _(cyclic dependency)_. In order to set these services and their helpers up in using clear dependency injection instead 0f being initiated in Repo inline.